compiler/pragmas.nim

```diff
@@ -691,14 +691,18 @@ proc singlePragma(c: PContext, sym: PSym, n: PNode, i: int,
   let ident = considerQuotedIdent(key)
   var userPragma = strTableGet(c.userPragmas, ident)
   if userPragma != nil:
-    inc c.instCounter
-    if c.instCounter > 100:
-      globalError(it.info, errRecursiveDependencyX, userPragma.name.s)
-    pragma(c, sym, userPragma.ast, validPragmas)
-    # ensure the pragma is also remember for generic instantiations in other
-    # modules:
-    n.sons[i] = userPragma.ast
-    dec c.instCounter
+    if userPragma.ast.len == 0:
+      delete(n.sons, i)
+      dec i
+    else:
+      inc c.instCounter
+      if c.instCounter > 100:
+        globalError(it.info, errRecursiveDependencyX, userPragma.name.s)
+      
+      pragma(c, sym, userPragma.ast, validPragmas)
+      n.sons[i..i] = userPragma.ast.sons
```

Comment:
I don't like these kind of AST transformations. Hard to review.

Reply:
Agree they are not straight forward, but I don't have better ideas. Let me know if you have some.
I will add comments to the code to make easier to follow.